### PR TITLE
KAFKA-5174: Have at least 2 threads

### PR DESCRIPTION
**Pull request by enothereska**
This fix needs to be backported to 0.10.2 as well.

### CODE REVIEW

**Comment by enothereska** [comment]
@dguy @guozhangwang please have a look. Note that this causes queryOnRebalance to also fail occasionally in 1 core environments.

**Review by dguy — APPROVED** [review]
LGTM

**Comment by enothereska** [comment]
queryOnRebalance continues to fail, will need more investigation after this PR. That investigation is part of https://github.com/apache/kafka/pull/2996

File: streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ public void openDB(ProcessorContext context) {
         // this is the recommended way to increase parallelism in RocksDb
         // note that the current implementation increases the number of compaction threads
         // but not flush threads.
-        options.setIncreaseParallelism(Runtime.getRuntime().availableProcessors());
+        options.setIncreaseParallelism(Math.max(Runtime.getRuntime().availableProcessors(), 2));
```
Comment (ijuma) [inline streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java]: Can you please add a comment explaining why the minimum is `2`?

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk.

**Comment by enothereska** [comment]
@guozhangwang also in 0.10.2 please. Thanks.

**Pull request closed without merging**